### PR TITLE
Rewrite unit tests for cluster module

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#4300|

Hi team,

This PR adds some new test for core/cluster and cluster/utils functions. In addition, multiples tests were fixed. More information here: #4300

### Black RBAC
```bash
adriiiprodri@wazuh pytest -vv test_rbac_black_cluster_endpoints.tavern.yaml
================================ test session starts ================================
platform linux -- Python 3.7.5, pytest-5.4.1, py-1.7.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.7.5', 'Platform': 'Linux-5.3.0-7648-generic-x86_64-with-Ubuntu-19.10-eoan', 'Packages': {'pytest': '5.4.1', 'py': '1.7.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '2.0.1', 'mock': '1.10.0', 'asyncio': '0.12.0', 'aiohttp': '0.3.0', 'cov': '2.8.1', 'testinfra': '5.0.0', 'tavern': '1.0.0', 'metadata': '1.8.0'}}
rootdir: /home/adriiiprodri/Desktop/git/wazuh/api/test/integration, inifile: pytest.ini
plugins: html-2.0.1, mock-1.10.0, asyncio-0.12.0, aiohttp-0.3.0, cov-2.8.1, testinfra-5.0.0, tavern-1.0.0, metadata-1.8.0
collected 21 items

test_rbac_black_cluster_endpoints.tavern.yaml::/cluster/local/config PASSED   [  4%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/healthcheck PASSED [  9%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/local/info PASSED [ 14%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/nodes PASSED      [ 19%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/nodes/{node_id} PASSED [ 23%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/status PASSED     [ 28%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration PASSED [ 33%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/configuration/validation PASSED [ 38%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration/{component}/{configuration} PASSED [ 42%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/info PASSED [ 47%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs PASSED [ 52%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs/summary PASSED [ 57%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats PASSED [ 61%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/status PASSED [ 66%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/files PASSED [ 71%]
test_rbac_black_cluster_endpoints.tavern.yaml::PUT /cluster/{node_id}/files PASSED [ 76%]
test_rbac_black_cluster_endpoints.tavern.yaml::DELETE /cluster/{node_id}/files PASSED [ 80%]
test_rbac_black_cluster_endpoints.tavern.yaml::GET /cluster/api/config PASSED [ 85%]
test_rbac_black_cluster_endpoints.tavern.yaml::PUT /cluster/api/config PASSED [ 90%]
test_rbac_black_cluster_endpoints.tavern.yaml::DELETE /cluster/api/config PASSED [ 95%]
test_rbac_black_cluster_endpoints.tavern.yaml::PUT /cluster/restart PASSED    [100%]

==================== 21 passed, 88 warnings in 156.19s (0:02:36) ====================
```
### White RBAC
```bash
adriiiprodri@wazuh feature/5109-JWT_invalidation ●  pytest -vv test_rbac_white_cluster_endpoints.tavern.yaml
================================ test session starts ================================
platform linux -- Python 3.7.5, pytest-5.4.1, py-1.7.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.7.5', 'Platform': 'Linux-5.3.0-7648-generic-x86_64-with-Ubuntu-19.10-eoan', 'Packages': {'pytest': '5.4.1', 'py': '1.7.0', 'pluggy': '0.13.1'}, 'Plugins': {'html': '2.0.1', 'mock': '1.10.0', 'asyncio': '0.12.0', 'aiohttp': '0.3.0', 'cov': '2.8.1', 'testinfra': '5.0.0', 'tavern': '1.0.0', 'metadata': '1.8.0'}}
rootdir: /home/adriiiprodri/Desktop/git/wazuh/api/test/integration, inifile: pytest.ini
plugins: html-2.0.1, mock-1.10.0, asyncio-0.12.0, aiohttp-0.3.0, cov-2.8.1, testinfra-5.0.0, tavern-1.0.0, metadata-1.8.0
collected 21 items

test_rbac_white_cluster_endpoints.tavern.yaml::/cluster/local/config PASSED   [  4%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/healthcheck PASSED [  9%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/local/info PASSED [ 14%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/nodes PASSED      [ 19%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/nodes/{node_id} PASSED [ 23%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/status PASSED     [ 28%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration PASSED [ 33%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/configuration/validation PASSED [ 38%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration/{component}/{configuration} PASSED [ 42%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/info PASSED [ 47%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs PASSED [ 52%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs/summary PASSED [ 57%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats PASSED [ 61%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/status PASSED [ 66%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/api/config PASSED [ 71%]
test_rbac_white_cluster_endpoints.tavern.yaml::PUT /cluster/api/config PASSED [ 76%]
test_rbac_white_cluster_endpoints.tavern.yaml::PUT /cluster/restart PASSED    [ 80%]
test_rbac_white_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/files PASSED [ 85%]
test_rbac_white_cluster_endpoints.tavern.yaml::PUT /cluster/{node_id}/files PASSED [ 90%]
test_rbac_white_cluster_endpoints.tavern.yaml::DELETE /cluster/{node_id}/files PASSED [ 95%]
test_rbac_white_cluster_endpoints.tavern.yaml::DELETE /cluster/api/config PASSED [100%]

==================== 21 passed, 90 warnings in 155.70s (0:02:35) ====================
```

Regards

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 84bf68699f57ce1bfa9c3f065f17826ef4eaad47
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2120/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2120/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2120/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2120.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2120/Analysis_dynamic_valgrind-v4.0.0-B2120.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 002069d1b448f36358596712de6e93b578778a3b
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2329/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2329/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2329/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2329.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2329/Analysis_dynamic_valgrind-v4.0.0-B2329.log)

File: framework/wazuh/core/cluster/tests/test_client.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                                                           cluster_items={'node': 'master-node', 'intervals': {
+                                                               'worker': {'connection_retry': 1}}},
+                                                           enable_ssl=False, performance_test=False,
+                                                           concurrency_test=False, file='None', string=20, logger=None)
+    abstract_client = client.AbstractClient(loop=loop, on_con_lost=None, name='Testing',
+                                            fernet_key='01234567891011121314151617181920',
+                                            logger=logger, manager=abstract_client_manager,
+                                            cluster_items={'node': 'master-node'})
+
+    result = await abstract_client_manager.start()
```
Comment (vicferpoy) [inline framework/wazuh/core/cluster/tests/test_client.py:34]: Can't we do anything with this `result` to improve the test?

File: framework/wazuh/core/cluster/tests/test_cluster.py [hunk 2]
```diff
@@ -26,44 +26,50 @@
...
+    'cluster': {
+        'key': 'a' * 32,
+        'node_name': 'master'
+    }
 }
 
 
 def test_read_empty_configuration():
     """
     Tests reading an empty cluster configuration
```
Comment (vicferpoy) [inline framework/wazuh/core/cluster/tests/test_cluster.py:66]:
```suggestion
    Test reading an empty cluster configuration.

```

File: framework/wazuh/core/cluster/tests/test_worker.py [hunk 3]
```diff
@@ -120,3 +133,130 @@ def test_remove_bulk_agents(isdir_mock, connection_mock, agents_mock, glob_mock,
...
+        assert worker_handler.process_request(command=b'dapi_res', data=b'Testing') == \
+               (b'ok', b'Response forwarded to worker')
+
+        # worker.py:162 local_server is not a member of AbstractClientManager
+        # assert worker_handler.process_request(command=b'dapi_err', data=b'Testing Error') == \
+        #        (b'ok', b'DAPI error forwarded to worker')
+
+        # worker.py:167 dapi is not a member of AbstractClientManager
+        # assert worker_handler.process_request(command=b'dapi', data=b'Testing Error') == \
+        #        (b'ok', b'Added request to API requests queue')
```
Comment (vicferpoy) [inline framework/wazuh/core/cluster/tests/test_worker.py:236]: Should this be commented?

Comment (AdriiiPRodri) [reply framework/wazuh/core/cluster/tests/test_worker.py:236]: This was commented and for the moment we are going to leave it until we define well the behavior of this module.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** afccc2a829aeb6bbfcc9df376bcce62cfc0af46e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2475/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2475/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2475/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2475.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2475/Analysis_dynamic_valgrind-v4.0.0-B2475.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 3549f9dc856de41909f3e6f09375c0b75f5af478
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2686/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2686/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2686/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2686.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2686/Analysis_dynamic_valgrind-v4.0.0-B2686.log)

**Comment by davidjiglesias** [comment]
### **Updated integration test results:**
```
test_cluster_endpoints.tavern.yaml 
	 56 passed, 72 warnings

test_rbac_black_cluster_endpoints.tavern.yaml 
	 21 passed, 23 warnings

test_rbac_white_cluster_endpoints.tavern.yaml 
	 21 passed, 23 warnings```

**Pull request merged**